test/qpy_compat/get_versions.py

```diff
@@ -0,0 +1,95 @@
+    # For more information, see:
+    # - https://packaging.python.org/en/latest/specifications/binary-distribution-format/
+    # - https://packaging.python.org/en/latest/specifications/platform-compatibility-tags/
+    #
+    # In particular, take note that a wheel's filename can include "compressed tag sets" (and our
+    # Linux wheels generally do this), which is why this function returns an iterable of tags.
+    _prefix, interpreters, abis, platforms = wheel[:-4].rsplit("-", 3)
+    yield from (
+        packaging.tags.Tag(interpreter, abi, platform)
+        for interpreter in interpreters.split(".")
+        for abi in abis.split(".")
+        for platform in platforms.split(".")
+    )
+
+
+def available_versions():
+    """Get all the versions of Qiskit that support exporting QPY, and are installable with the
+    active version of Python on this platform."""
+    our_version = packaging.version.parse(qiskit.__version__)
+    supported_tags = set(packaging.tags.sys_tags())
+
+    def available_versions_for_package(package, min_version=None, max_version=None):
+        with urllib.request.urlopen(f"https://pypi.org/pypi/{package}/json") as fd:
+            data = json.load(fd)
+        min_version = min_version and packaging.version.parse(min_version)
+        max_version = max_version and packaging.version.parse(max_version)
+        for other_version, payload in data["releases"].items():
+            other_version = packaging.version.parse(other_version)
+            if min_version is not None and other_version < min_version:
+                continue
+            if max_version is not None and other_version >= max_version:
+                continue
+            if other_version > our_version or other_version.is_prerelease:
+                continue
+            # Note: this ignores versions that are uninstallable because we're using a Python
+            # version that's too new, which can be a problem for the oldest Terras, especially from
+            # before we built for abi3.  We're not counting sdists, since if we didn't release a
+            # compatible wheel for the current Python version, there's no guarantee it'll install.
+            if not any(
+                tag in supported_tags
+                for release in payload
+                if release["packagetype"] == "bdist_wheel"
+                for tag in tags_from_wheel_name(release["filename"])
+            ):
+                print(
+                    f"skipping '{other_version}', which has no installable binary artifacts",
+                    file=sys.stderr,
+                )
+                continue
+            yield other_version
+
+    yield from (
+        ("qiskit-terra", version)
+        for version in available_versions_for_package("qiskit-terra", "0.18.0", "1.0.0")
+    )
+    yield from (
+        ("qiskit", version) for version in available_versions_for_package("qiskit", "1.0.0")
+    )
+
+
+def main():
+    """main"""
```

Comment:
Hah, I like this docstring

Reply:
passive aggressive suppression of the pylint warning haha